### PR TITLE
Add missing win32 api related to icons and menus.

### PR DESCRIPTION
**Pull request by lordhippo**
These are the missing APIs I discovered while working on https://github.com/lordhippo/odin-tray

### CODE REVIEW

File: core/sys/windows/user32.odin [hunk 1]
```diff
@@ -489,3 +500,31 @@ WINDOWINFO :: struct {
...
+MIIM_CHECKMARKS :: 0x00000008
+MIIM_TYPE       :: 0x00000010
+MIIM_DATA       :: 0x00000020
+
+MIIM_STRING :: 0x00000040
+MIIM_BITMAP :: 0x00000080
+MIIM_FTYPE  :: 0x00000100
+
+MENUITEMINFOW :: struct
+{
```
Comment (gingerBill) [inline core/sys/windows/user32.odin:516]: Only gripe: make sure to use 1TBS style braces: `struct {`

File: core/sys/windows/user32.odin [hunk 2]
```diff
@@ -489,3 +500,30 @@ WINDOWINFO :: struct {
...
+    fType: UINT,         // used if MIIM_TYPE (4.0) or MIIM_FTYPE (>4.0)
+    fState: UINT,        // used if MIIM_STATE
+    wID: UINT,           // used if MIIM_ID
+    hSubMenu: HMENU,      // used if MIIM_SUBMENU
+    hbmpChecked: HBITMAP,   // used if MIIM_CHECKMARKS
+    hbmpUnchecked: HBITMAP, // used if MIIM_CHECKMARKS
+    dwItemData: ULONG_PTR,   // used if MIIM_DATA
+    dwTypeData: LPWSTR,    // used if MIIM_TYPE (4.0) or MIIM_STRING (>4.0)
+    cch: UINT,           // used if MIIM_TYPE (4.0) or MIIM_STRING (>4.0)
+    hbmpItem: HBITMAP,      // used if MIIM_BITMAP
```
Comment (gingerBill) [inline core/sys/windows/user32.odin:527]: Make sure to use tabs rather than spaces.

Comment (Hyrtwol) [reply core/sys/windows/user32.odin:527]: Try to rebase up I think most of the changes here are present in master now ;)

**Comment by Hyrtwol** [comment]
Try to rebase up I think most of the changes here are present in master now ;)

**Pull request closed without merging**